pkgs/by-name/tp/tpm2-openssl/package.nix

```
@@ -0,0 +1,51 @@
+{ stdenv
+, lib
+, autoreconfHook
+, fetchFromGitHub
+, autoconf-archive
+, pkg-config
+, openssl
+, tpm2-tss
+,
+}:
+stdenv.mkDerivation (
+  finalAttrs: {
+    pname = "tpm2-openssl";
+    version = "1.2.0";
+    src = fetchFromGitHub {
+      owner = "tpm2-software";
+      repo = "tpm2-openssl";
+      rev = finalAttrs.version;
+      hash = "sha256-mZ4Z/GxJFwwfyFd1SAiVlQqOjkFSzsZePeuEZtq8Mcg=";
+    };
+
+    nativeBuildInputs = [
+      autoreconfHook
+      autoconf-archive
+      pkg-config
+    ];
+
+    buildInputs = [
+      openssl
+      tpm2-tss
+    ];
+
+    preConfigurePhases = [ "setConfigureFlags" ];
+
+    env.setConfigureFlags = ''
+      export configureFlags="$configureFlags --with-modulesdir=$out/lib/ossl-modules"
+    '';
```

```suggestion
    preConfigure = ''
      export configureFlags="$configureFlags --with-modulesdir=$out/lib/ossl-modules"
    '';
```
we really don't need to invent a new phase here

---

Oh sorry, I am still a newbie in that regard :D

Fixed in https://github.com/NixOS/nixpkgs/commit/0321a11faeffe51312ba34d8d93b436c59d570ca